### PR TITLE
Upload file token issue #150

### PR DESCRIPTION
**Pull request by emli**
Please review my PR. @dilshat @crioto

### CODE REVIEW

**Comment by Dilshat** [comment]
This merge requires changes in agent and hub. So let's merge when  we are ready there too.

File: upload/upload.go [hunk 1]
```diff
@@ -28,15 +28,18 @@ type share struct {
...
+	if len(token) == 0 || len(db.CheckToken(token)) == 0 {
 		w.WriteHeader(http.StatusUnauthorized)
 		w.Write([]byte("Not authorized"))
 		log.Warn(r.RemoteAddr + " - rejecting unauthorized upload request")
 		return
 	}
+	r.ParseMultipartForm(32 << 20)
 
-	owner = strings.ToLower(db.CheckToken(r.MultipartForm.Value["token"][0]))
+	owner = strings.ToLower(db.CheckToken(token))
```
Comment (Dilshat) [inline upload/upload.go]: @emli why you perform call to db.CheckToken(token) 2 times : here and on line 34
better optimize this to have only one call by storing result of the first call in a variable

File: upload/upload.go [hunk 2]
```diff
@@ -28,15 +28,18 @@ type share struct {
 
 //Handler function works with income upload requests, makes sanity checks, etc
 func Handler(w http.ResponseWriter, r *http.Request) (md5sum, sha256sum, owner string) {
-	r.ParseMultipartForm(32 << 20)
-	if len(r.MultipartForm.Value["token"]) == 0 || len(db.CheckToken(r.MultipartForm.Value["token"][0])) == 0 {
+
```
[the remark this answers is not in the document]
Comment (emli) [reply upload/upload.go]: Fixed

File: upload/upload.go (hunk 1 above)
Comment (emli) [reply upload/upload.go]: fixed

**Pull request merged**